### DISCUSSION TITLE
App Hangs on SteamDL for PalWorld Server

### DISCUSSION THREAD

**Discussion by maneumann-it** [pufferpanel/pufferpanel#1378] [Q&A] [answered]
Good day, 

I have a PalWorld server I deployed successfully with PufferPanel (on an Asustor NAS). Well, a big update came by and I thought I would restart the server to download the latest update and now it is stuck trying to get information. I don't know if it is a PufferPanel issue or DepotDownloader, but something isn't working.

The exact step (for those on search engines looking for this) is: Got AppInfo for 2394010

Some troubleshooting steps I have tried:
- Killing and restarting the server in the PufferPanel UI
- Restarting PufferPanel via Portainer.
- Restarting PufferPanel via App Central in the Asustor UI
- Deleting the .depotdownloader folder on the server
- Deleting the .steam folder on the server
- Restarting the NAS
- Deploying a fresh server (same issue happens)
- Just waiting (eventually times out)
- Deleting/Renaming the DepotDownloader folder in PufferPanel's docker > binaries folder.

![PufferPanel_PalWorld_Server_Hang](https://github.com/user-attachments/assets/8cc4c757-4cad-490f-9f0d-841bcbe59a3d)
[5c880fc6.json](https://github.com/user-attachments/files/18201337/5c880fc6.json)
[config (1).json](https://github.com/user-attachments/files/18201338/config.1.json)

Any advice or suggestions?

I did have to make some adjustments to the PalWorld Config to make it deploy correctly. The config loaded by default is outdated (a bug I will submit at a later date).

**Comment by LordRalex** [comment, marked as answer]
It is in fact DepotDownloader.

```bash
curl https://github.com/SteamRE/DepotDownloader/releases/download/DepotDownloader_2.7.4/DepotDownloader-linux-x64.zip -L -o /tmp/dd.zip 
unzip /tmp/dd.zip -d /tmp/dd
sudo rm /var/lib/pufferpanel/binaries/depotdownloader/DepotDownloader
sudo mv /tmp/dd/DepotDownloader /var/lib/pufferpanel/binaries/depotdownloader/DepotDownloader 
sudo chown pufferpanel:pufferpanel /var/lib/pufferpanel/binaries/depotdownloader/DepotDownloader 
rm -r /tmp/dd.zip /tmp/dd
```

**Reply by maneumann-it** [reply]
Thank you so much! That absolutely fixed it!

**Comment by cooler0918** [comment]
Thank you, that did it!

**Comment by adroslice** [comment]
To people in the future who might have an issue after following the marked answer, make sure to update the download link to the latest version of DepotDownloader!

**Reply by maneumann-it** [reply]
This fixed another issue for me!

**Reply by cowchmonkey** [reply]
> To people in the future who might have an issue after following the marked answer, make sure to update the download link to the latest version of DepotDownloader!

worked for me as well :D thanks bunches
